src/sentry/utils/event_frames.py

```diff
@@ -41,6 +41,10 @@ class SdkFrameMunger:
 def java_frame_munger(frame: EventFrame) -> str | None:
     if not frame.filename or not frame.module:
         return None
+
+    if "$$" in frame.module:
+        return frame.module.split("$$")[0].replace(".", "/") + ".java"
```

Comment:
I guess this can be also kotlin right (or any other jvm lang)? if we use event.platform for this munger then all of those jvm platforms will have `java`, but the file extension can be different

Reply:
@romtsn it's based on the `platform`. It doesn't matter what the SDK name is.

If the file does not end with `.java` it won't work.

This is only useful because Source Context is not available.

If you want this to work for other extensions, let me know if there's a way to determine the file extension.

Reply:
@armenzg can't we search for files on GitHub without the extension?
This guessing game is why we didn't include the file extension on the source ZIP but instead renamed everything to `.jvm`.

I'd say Kotlin is very important.
Other's we could also consider are Groovy, Scala, Clojure, JRuby.

All of these are in use by our customers, but I can't say whether they write their own code in any of those languages or if it's just their dependencies that use them. For Kotlin we do know for certain our customers are using it directly since it's very popular on Android but it's also used on Backends.

There's also Jython.

Reply:
IIRC the file extension isn't always there / correct.

Reply:
Maybe the "Implementation Details" section on this comment is helpful: https://github.com/getsentry/sentry-java/issues/633#issuecomment-1465599120

Reply:
oh now I remember.. I guess we should replicate the same logic for defining the file path as we've done for source contexts

Reply:
I chose `.java` as the most obvious extension to me. This only happens when the stack trace looks like this:

```
filename="D8$$SyntheticClass",
module="com.example.vu.android.MainActivity$$ExternalSyntheticLambda4",
```

If you give me a short list of extensions, we could have a hack where for Java we try a few extensions.